src/wallet/wallettool.cpp

```diff
@@ -90,11 +92,12 @@ static std::shared_ptr<CWallet> LoadWallet(const std::string& name, const fs::pa
 
 static void WalletShowInfo(CWallet* wallet_instance)
 {
-    LOCK(wallet_instance->cs_wallet);
+    auto spk_man = wallet_instance->GetLegacyScriptPubKeyMan();
+    LOCK2(wallet_instance->cs_wallet, spk_man->cs_KeyStore);
 
     tfm::format(std::cout, "Wallet info\n===========\n");
     tfm::format(std::cout, "Encrypted: %s\n", wallet_instance->IsCrypted() ? "yes" : "no");
-    tfm::format(std::cout, "HD (hd seed available): %s\n", wallet_instance->GetHDChain().seed_id.IsNull() ? "no" : "yes");
+    tfm::format(std::cout, "HD (hd seed available): %s\n", (spk_man && spk_man->GetHDChain().seed_id.IsNull()) ? "no" : "yes");
```

Comment:
is checking that `spk_man` isn't null needed here assuming you already call `spk_man->cs_KeyStore` which is a deference?

Reply:
&& short-circuits; it doesn't evaluate its right hand is the left hand was false.

Reply:
Yes. My point was that he deferences it a few lines above when he acquires a lock.

Reply:
wallettool looks to be broken when there are other `ScriptPubKeyMan` classes. I'll investigate in the Descriptor wallets PR.

Reply:
This change to using `spk_man` here is not actually needed. I've changed it to use `CWallet::IsHDEnabled`